datafusion/expr-common/src/statistics.rs

```diff
@@ -857,6 +857,143 @@ pub fn compute_variance(
     ScalarValue::try_from(target_type)
 }
 
+/// Merges two distributions into a single distribution that represents their combined statistics.
+/// This creates a more general distribution that approximates the mixture of the input distributions.
+pub fn merge_distributions(a: &Distribution, b: &Distribution) -> Result<Distribution> {
```

Comment:
I wonder if this should be a method on `Distribution` rather than a free function 🤔

Reply:
Yes, that's also what I wanted to do